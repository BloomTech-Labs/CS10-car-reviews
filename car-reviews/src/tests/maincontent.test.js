import MainContent from '../components/MainPage/maincontent';

describe('MainContent', () => {
  const wrapper = () => shallow(<MainContent />);

  // All tests will go here
  it('renders MainContent component correctly', () => {
    expect(wrapper()).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const divs = wrapper().find('div');
    expect(divs.length).toBeGreaterThan(0);
  });

  describe('the rendered div', () => {
    it('contains everything else that gets rendered', () => {
      const divs = wrapper().find('div');
      const wrappingDiv = divs.first();

      expect(wrappingDiv.children()).toEqual(wrapper().children());
    });
  });

  it('always renders a Container', () => {
    const container = wrapper().find('Container');
    expect(container.length).toBe(1);
  });

  describe('the rendered Container', () => {
    it('contains everything else that gets rendered', () => {
      const container = wrapper().find('Container');
      const wrappingContainer = container.first();

      expect(wrappingContainer.children()).toEqual(container.children());
    });
  });

  describe('Row component', () => {
    it('should render exactly three <Row/> components', () => {
      const rows = wrapper().find('Row');
      expect(rows.length).toEqual(3);
    });
  });

  describe('h3 elements', () => {
    it('should render exactly three <h3/> elements', () => {
      const headers = wrapper().find('h3');
      expect(headers.length).toEqual(3);
    });
  });

  describe('Testing state of MainContent', () => {
    it('counter is defaulted to 0', () => {
      expect(wrapper().state('counter')).toBe(0);
    });

    it('popularCars array is empty before CDM()', () => {
      expect(wrapper().state('popularCars')).toHaveLength(0);
    });

    it('reviews array is empty before CDM()', () => {
      expect(wrapper().state('popularCars')).toHaveLength(0);
    });

    it('popularReviewers array is empty before CDM()', () => {
      expect(wrapper().state('popularCars')).toHaveLength(0);
    });

    it('newdate defaulted to dateString', () => {
      const currentDate = new Date();
      const date = currentDate.getDate();
      const month = currentDate.getMonth();
      const year = currentDate.getFullYear();
      const dateString = date + '-' + (month + 1) + '-' + year;

      expect(wrapper().state('newdate')).toBe(dateString);
    });

    it('newdate defaulted to dateString', () => {
      const currentDate = new Date();
      const date = currentDate.getDate();
      const month = currentDate.getMonth();
      const year = currentDate.getFullYear();
      const dateString = date + '-' + (month + 1) + '-' + year;

      expect(wrapper().state('newdate')).toBe(dateString);
    });
  });
});
