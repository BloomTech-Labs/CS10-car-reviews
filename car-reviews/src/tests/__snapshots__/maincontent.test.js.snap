// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent renders MainContent component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="main-content"
>
          <div
                    style={
                              Object {
                                        "height": "20px",
                                      }
                    }
          />
          <Container
                    tag="div"
          >
                    <h3
                              className="header"
                    >
                              Featured Reviews
                    </h3>
                    <Row
                              tag="div"
                    >
                              
                    </Row>
                    <h3
                              className="header"
                    >
                              Popular Cars
                    </h3>
                    <Row
                              tag="div"
                    >
                              
                    </Row>
                    <div
                              style={
                                        Object {
                                                  "height": "50px",
                                                }
                              }
                    />
                    <h3
                              className="header"
                    >
                              Popular Reviewers
                    </h3>
                    <Row
                              style={
                                        Object {
                                                  "marginTop": "30px",
                                                }
                              }
                              tag="div"
                    >
                              
                    </Row>
          </Container>
          <div
                    style={
                              Object {
                                        "height": "100px",
                                      }
                    }
          />
</div>,
      ],
      "className": "main-content-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                            Object {
                                          "height": "20px",
                                        }
              }
/>,
            <Container
              tag="div"
>
              <h3
                            className="header"
              >
                            Featured Reviews
              </h3>
              <Row
                            tag="div"
              >
                            
              </Row>
              <h3
                            className="header"
              >
                            Popular Cars
              </h3>
              <Row
                            tag="div"
              >
                            
              </Row>
              <div
                            style={
                                          Object {
                                                        "height": "50px",
                                                      }
                            }
              />
              <h3
                            className="header"
              >
                            Popular Reviewers
              </h3>
              <Row
                            style={
                                          Object {
                                                        "marginTop": "30px",
                                                      }
                            }
                            tag="div"
              >
                            
              </Row>
</Container>,
            <div
              style={
                            Object {
                                          "height": "100px",
                                        }
              }
/>,
          ],
          "className": "main-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "style": Object {
                "height": "20px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <h3
                  className="header"
>
                  Featured Reviews
</h3>,
                <Row
                  tag="div"
>
                  
</Row>,
                <h3
                  className="header"
>
                  Popular Cars
</h3>,
                <Row
                  tag="div"
>
                  
</Row>,
                <div
                  style={
                                    Object {
                                                      "height": "50px",
                                                    }
                  }
/>,
                <h3
                  className="header"
>
                  Popular Reviewers
</h3>,
                <Row
                  style={
                                    Object {
                                                      "marginTop": "30px",
                                                    }
                  }
                  tag="div"
>
                  
</Row>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Featured Reviews",
                  "className": "header",
                },
                "ref": null,
                "rendered": "Featured Reviews",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Popular Cars",
                  "className": "header",
                },
                "ref": null,
                "rendered": "Popular Cars",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "style": Object {
                    "height": "50px",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Popular Reviewers",
                  "className": "header",
                },
                "ref": null,
                "rendered": "Popular Reviewers",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [],
                  "style": Object {
                    "marginTop": "30px",
                  },
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "style": Object {
                "height": "100px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="main-content"
>
            <div
                        style={
                                    Object {
                                                "height": "20px",
                                              }
                        }
            />
            <Container
                        tag="div"
            >
                        <h3
                                    className="header"
                        >
                                    Featured Reviews
                        </h3>
                        <Row
                                    tag="div"
                        >
                                    
                        </Row>
                        <h3
                                    className="header"
                        >
                                    Popular Cars
                        </h3>
                        <Row
                                    tag="div"
                        >
                                    
                        </Row>
                        <div
                                    style={
                                                Object {
                                                            "height": "50px",
                                                          }
                                    }
                        />
                        <h3
                                    className="header"
                        >
                                    Popular Reviewers
                        </h3>
                        <Row
                                    style={
                                                Object {
                                                            "marginTop": "30px",
                                                          }
                                    }
                                    tag="div"
                        >
                                    
                        </Row>
            </Container>
            <div
                        style={
                                    Object {
                                                "height": "100px",
                                              }
                        }
            />
</div>,
        ],
        "className": "main-content-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                                Object {
                                                "height": "20px",
                                              }
                }
/>,
              <Container
                tag="div"
>
                <h3
                                className="header"
                >
                                Featured Reviews
                </h3>
                <Row
                                tag="div"
                >
                                
                </Row>
                <h3
                                className="header"
                >
                                Popular Cars
                </h3>
                <Row
                                tag="div"
                >
                                
                </Row>
                <div
                                style={
                                                Object {
                                                                "height": "50px",
                                                              }
                                }
                />
                <h3
                                className="header"
                >
                                Popular Reviewers
                </h3>
                <Row
                                style={
                                                Object {
                                                                "marginTop": "30px",
                                                              }
                                }
                                tag="div"
                >
                                
                </Row>
</Container>,
              <div
                style={
                                Object {
                                                "height": "100px",
                                              }
                }
/>,
            ],
            "className": "main-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "style": Object {
                  "height": "20px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <h3
                    className="header"
>
                    Featured Reviews
</h3>,
                  <Row
                    tag="div"
>
                    
</Row>,
                  <h3
                    className="header"
>
                    Popular Cars
</h3>,
                  <Row
                    tag="div"
>
                    
</Row>,
                  <div
                    style={
                                        Object {
                                                            "height": "50px",
                                                          }
                    }
/>,
                  <h3
                    className="header"
>
                    Popular Reviewers
</h3>,
                  <Row
                    style={
                                        Object {
                                                            "marginTop": "30px",
                                                          }
                    }
                    tag="div"
>
                    
</Row>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Featured Reviews",
                    "className": "header",
                  },
                  "ref": null,
                  "rendered": "Featured Reviews",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [],
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Popular Cars",
                    "className": "header",
                  },
                  "ref": null,
                  "rendered": "Popular Cars",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [],
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "style": Object {
                      "height": "50px",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Popular Reviewers",
                    "className": "header",
                  },
                  "ref": null,
                  "rendered": "Popular Reviewers",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [],
                    "style": Object {
                      "marginTop": "30px",
                    },
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "style": Object {
                  "height": "100px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
